src/serve/proxy.rs

```diff
@@ -62,7 +62,7 @@ impl ProxyBuilder {
         } else {
             let no_sys_proxy = opts.no_system_proxy;
             let insecure = opts.insecure;
-            let client = self.clients.get_client(opts)?;
+            let client = self.clients.get_client(opts.clone())?;
```

Comment:
If you'd pull the tracing statement  above this line, you wouldn't need to clone.